### PR TITLE
Initialization of  Default Protocol class

### PR DESCRIPTION
**Pull request by rasswanth-s**
## Description
Initialization of the Default Protocol class.The default protocol will contain all the primtive operations.In a session if a given protocol does not implement a specific operation.It would be reverted to the Default Protocol class.

## Affected Dependencies
Since Falcon is 3PC,Adding DefaultProtocol would make RSTensor framework agnostic

## How has this been tested?
Added tests for the same.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/sympc/protocol/default/default.py
Comment (gmuraru) [inline src/sympc/protocol/default/default.py:17]: Q: Could we have it `ShareTensor` (because all the operations that work are using `ShareTensor`)

File: src/sympc/protocol/default/default.py
Comment (gmuraru) [inline src/sympc/protocol/default/default.py:35]: Ideally, after we merge this PR, we could create another where we put all the `add`, `sub`, `mult`, `matmul`, `eq` .... operation here

**Review by gmuraru — APPROVED** [review]
LGTM!  Let's wait for another team member to review this before merging

File: src/sympc/protocol/default/default.py
Comment (gmuraru) [reply src/sympc/protocol/default/default.py:17]: And also the `security_levels` to be only `["semi-honest"]`

Comment (rasswanth-s) [reply src/sympc/protocol/default/default.py:17]: Yeah ,george  I wanted to initialize `default protocol` ,such that I could revert `RSTensor` operation when we have arbitrary number of parties,as `Falcon` is highly customized for 3PC.Shall we have it as `RSTensor` for now?.In the long run when we are implementing ops we could have both `RSTensor` and `ShareTensor`.

**Review by kamathhrishi — APPROVED** [review]

**Review by kamathhrishi — COMMENTED** [review]
Will be revisiting this PR after a while.

**Comment by kamathhrishi** [comment]
We can close and come back to the PR later?

**Pull request closed without merging**
